### PR TITLE
Unhide newer use case revisions

### PR DESCRIPTION
**Pull request by gnidan**
Previously I revised the "Use Cases" section, but the newer revisions never made it to the document tree.

- Removes the currently-pointed-at older revision and points to the newer revision.
- Rename use cases to be Truffle/tooling specific, to be more clear that these are not specifically the use cases of Truffle DB
- Also bind the livehtml server on 0.0.0.0 (because I'm using two computers these days)

[Preview (direct link to "Requirements")](https://trufflesuite.github.io/artifact-updates/contrib/trufflesuite/unhide/requirements/index.html)

### CODE REVIEW

File: docs/use-cases.rst [hunk 1]
```diff
@@ -1,501 +0,0 @@
...
-```````````````````````````````
-
-Similarly to calls, most contracts provide interface methods for modifying
-|ContractInstance| state. Users must be able to execute the allowable
-"write" operations for a contract instance.
-
-Send ETH to a contract instance
-```````````````````````````````
-
-Many smart contracts expose methods for receiving payment in the form or ether
```
Comment (fainashalts) [inline docs/use-cases.rst]: This should be "in the form *of* ether"

File: docs/use-cases.rst [hunk 2]
```diff
@@ -1,501 +0,0 @@
...
-or other tokens.
-
-Developers must be able to test the receipt of ether; business stakeholders may
-have to provide initial funds, for applications that require it.
```
Comment (fainashalts) [inline docs/use-cases.rst]: remove comma after funds I think

File: docs/use-cases.rst (hunk 1 above)
Comment (gnidan) [reply docs/use-cases.rst]: Well, that file was deleted, so I think the typo no longer exists :)

**Comment by fainashalts** [comment]
LGTM!

**Comment by gnidan** [comment]
Thanks @fainashalts !

**Comment by gnidan** [comment]
I'm impatient, merging without Travis's stamp of approval.

**Pull request merged**